### ISSUE TITLE
T_DATA warning from extension on Ruby 3.2

### ISSUE THREAD

**Issue by nickcampbell18** [appsignal/appsignal-ruby#909]
## Describe the bug

After upgrading to Ruby 3.2, I'm getting a sporadic warning from the Appsignal extension:

```
$ rails test
Running 16 tests in a single process (parallelization threshold is 50)
Run options: --seed 11057

# Running:

/Users/nickcampbell/.asdf/installs/ruby/3.2.0/lib/ruby/gems/3.2.0/gems/appsignal-3.3.0/lib/appsignal/utils/data.rb:19: warning: undefining the allocator of T_DATA class Appsignal::Extension::Data
................
```

Fortunately it doesn't seem to crash the process etc, so it's just an annoyance for now. When googling I found this similar issue on another repo, if it helps: https://github.com/rails-sqlserver/tiny_tds/issues/515

## To Reproduce

Here's my environment:

```sh
$ ruby -v
ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-darwin22]

$ bundle info appsignal
  * appsignal (3.3.0)
```

This is the line of Ruby that it's talking about:

```ruby
map = Appsignal::Extension.data_map_new
```

**Comment by tombruijn** [comment]
Thanks for the report @nickcampbell18! We'll have a look what's causing this. Doesn't appear to be directly something we do, but most likely it's from our C-extension based on the warning origin.

**Comment by edouard** [comment]
The Redcarpet gem also has that warning on ruby 3.2. Here is a pull request showing how they fixed it if that's of any help: https://github.com/vmg/redcarpet/pull/721/

**Comment by inductor** [comment]
redcarpet 3.6.0 is out FYI

**Comment by tombruijn** [comment]
I published the fix for this in Ruby gem 3.3.3 🚀
